File: exercise/scala/day03/src/main/scala/preparation/SantaWorkshopService.scala
package preparation

class SantaWorkshopService {
  private val preparedGifts: scala.collection.mutable.ListBuffer[Gift] = scala.collection.mutable.ListBuffer()

  def prepareGift(giftName: String, weight: Double, color: String, material: String): Gift = {
    if (weight > 5) {
      throw new IllegalArgumentException("Gift is too heavy for Santa's sleigh")
    }

    val gift = new Gift(giftName, weight, color, material)
    preparedGifts += gift

    gift
  }
}

File: exercise/kotlin/day03/src/main/kotlin/preparation/SantaWorkshopService.kt
package preparation

class SantaWorkshopService {
    private val preparedGifts = mutableListOf<Gift>()

    fun prepareGift(giftName: String, weight: Double, color: String, material: String): Gift {
        if (weight > 5) {
            throw IllegalArgumentException("Gift is too heavy for Santa's sleigh")
        }

        val gift = Gift(giftName, weight, color, material)
        preparedGifts.add(gift)

        return gift
    }
}